tests/store/test_sftp_artifact_repo.py

```diff
@@ -0,0 +1,100 @@
+from mock import Mock, MagicMock
+import unittest
+
+import pysftp
+
+from mlflow.store.artifact_repo import ArtifactRepository
+from mlflow.store.sftp_artifact_repo import SFTPArtifactRepository
+
+
+class SFTPArtifactRepositryTest(unittest.TestCase):
```

Comment:
Would it be possible to add tests for `log_artifact` and `log_artifacts`? e.g. we could try logging artifacts to `sftp://<tempdir>` and verify they appear where expected. If such tests require SSH configuration, we could mark them as such via `pytest.mark.requires_ssh` (see [projects CLI tests](https://github.com/mlflow/mlflow/blob/HEAD/tests/projects/test_projects_cli.py#L39))